webapp/channels/src/components/post_view/message_attachments/action_button/action_button.tsx

```diff
@@ -34,18 +37,20 @@ export default class ActionButton extends React.PureComponent<Props> {
     render() {
         const {action, handleAction, disabled, theme} = this.props;
         let customButtonStyle;
+        const actionStyle = action.style || 'primary';
```

Comment:
This is with `default` and `centerChannelBg`

[Screencast from 29-07-23 12:07:01 AM IST.webm](https://github.com/mattermost/mattermost/assets/83122406/c760e5ca-98d8-4a4d-a069-ddca17a53f71)

This is with `default` and `centerChannelColor`

[Screencast from 29-07-23 12:05:45 AM IST.webm](https://github.com/mattermost/mattermost/assets/83122406/057999a2-0bc7-4517-b14b-115aa7255fbf)

Both personally didn't appealed to me so I went with `primary`.

Also as you mentioned earlier instead of `color`, `backgroundcolor` is being applied, something I thought of bringing notice to again.

And never of these above two styles and the one with 'primary' show any hover state behavior. 
[Hover State Reference](https://github.com/mattermost/mattermost/issues/23968#issuecomment-1652463700)

Reply:
@mickmister which style should I go with?

Reply:
Hi @sehgxl, I'm thinking `primary` looks good here (the gif shown in the PR description). @matthewbirtch What are your thoughts on this?

Reply:
I agree `primary` looks good, but don't we need to be able to support themes for both `primary` _and_ `default` style? Wasn't the default style button the one that is the problem for theming here?

Reply:
@matthewbirtch There were a few issues here:

- If the integration provided no color property for the button, there was no styling/theming being done to the button (The rest of items below assume the integration provided a color). May be good to revisit this holistically as the global css of the application doesn't seem to gracefully handle buttons without explicit styling.
- We were setting the `color` of the button instead of the `background-color`, resulting in essentially no styling of the button itself and instead only the button's text.
- `default` option was using `centerChannelColor` instead of `centerChannelBg`, which results in the white background for the button yet again.

I believe both `primary` and `default` styling are themed properly now. The remaining topic is which one should be used when the integration does not provide a color option for the button.

Reply:
Thanks for filling me in on the context @mickmister, that's the part of the story I was missing.

I agree we probably want to revisit buttons holistically - both from a consistent styling perspective, hover/focus states, and handling the cases you mentioned. This gets back to the conversation about updating these interactive components to match the latest design styling, but I think it would be great to make updates here. Maybe we should create a ticket to at least have it queued up in Jira.

As for which one we should use when the integration doesn't provide a color, let's go with primary.